LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ModuleConnection/ModuleConnectionHost.cs

```diff
@@ -0,0 +1,211 @@
+                throw;
+            }
+
+            var moduleTwinCollection = moduleTwin.Properties.Desired;
+
+            if (!TryUpdateConfigurationWithDesiredProperties(moduleTwinCollection))
+            {
+                Logger.Log($"The initial configuration of the facade function could not be found in the desired properties", LogLevel.Error);
+                throw new ConfigurationErrorsException("Could not get Facade information from module twin");
+            }
+
+            await this.loRaModuleClient.SetDesiredPropertyUpdateCallbackAsync(OnDesiredPropertiesUpdate, null);
+
+            await this.loRaModuleClient.SetMethodDefaultHandlerAsync(OnDirectMethodCalled, null);
+        }
+
+        internal async Task<MethodResponse> OnDirectMethodCalled(MethodRequest methodRequest, object userContext)
+        {
+            if (methodRequest == null) throw new ArgumentNullException(nameof(methodRequest));
+
+            if (string.Equals(Constants.CloudToDeviceClearCache, methodRequest.Name, StringComparison.OrdinalIgnoreCase))
+            {
+                return await ClearCache();
+            }
+            else if (string.Equals(Constants.CloudToDeviceDecoderElementName, methodRequest.Name, StringComparison.OrdinalIgnoreCase))
+            {
+                return await SendCloudToDeviceMessageAsync(methodRequest);
+            }
+
+            Logger.Log($"Unknown direct method called: {methodRequest.Name}", LogLevel.Error);
+
+            return new MethodResponse((int)HttpStatusCode.NotFound);
+        }
+
+        internal async Task<MethodResponse> SendCloudToDeviceMessageAsync(MethodRequest methodRequest)
+        {
+            if (!string.IsNullOrEmpty(methodRequest.DataAsJson))
+            {
+                ReceivedLoRaCloudToDeviceMessage c2d = null;
+
+                try
+                {
+                    c2d = JsonSerializer.Deserialize<ReceivedLoRaCloudToDeviceMessage>(methodRequest.DataAsJson);
+                }
+                catch (JsonException ex)
+                {
+                    Logger.Log($"Impossible to parse Json for c2d message for device {c2d?.DevEUI}, error: {ex}", LogLevel.Error);
+                    return new MethodResponse((int)HttpStatusCode.BadRequest);
+                }
+
+
+                Logger.Log(c2d.DevEUI, $"received cloud to device message from direct method: {methodRequest.DataAsJson}", LogLevel.Debug);
+
+                using var cts = methodRequest.ResponseTimeout.HasValue ? new CancellationTokenSource(methodRequest.ResponseTimeout.Value) : null;
+
+                if (await this.classCMessageSender.SendAsync(c2d, cts?.Token ?? CancellationToken.None))
+                {
+                    return new MethodResponse((int)HttpStatusCode.OK);
+                }
+            }
+
+            return new MethodResponse((int)HttpStatusCode.BadRequest);
```

Comment:
In `OnDirectMethodCalled`, we return 404, but here we return 400. Any reason for the discrepancy?

Reply:
No idea, this was the logic before I will put everything to 400